src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.cs

```diff
@@ -12,37 +14,194 @@ internal static partial class ClrThreadPool
         
         private static int s_minThreads; // TODO: Initialize
         private static int s_maxThreads; // TODO: Initialize
+        private static readonly LowLevelLock s_maxMinThreadLock = new LowLevelLock();
+        
+        private static readonly LowLevelLock s_threadAdjustmentLock = new LowLevelLock();
+        private static ThreadCounts s_counts = new ThreadCounts();
+        private static int s_lastDequeueTime;
+        private static int s_priorCompletedWorkRequestsTime;
+        private static int s_nextCompletedWorkRequestsTime;
+        private static long s_currentSampleStartTime;
+        private static int s_priorCompletionCount = 0;
+        private static int s_completionCount = 0;
+        private static int s_threadAdjustmentInterval;
 
-        // TODO: SetMinThreads and SetMaxThreads need to be synchronized with a lock
-        // TODO: Compare with CoreCLR implementation and ensure this has the same guarantees.
         public static bool SetMinThreads(int threads)
         {
+            s_maxMinThreadLock.Acquire();
+            bool success;
             if (threads < 0 || threads > s_maxThreads)
             {
-                return false;
+                success = false;
             }
             else
             {
                 s_minThreads = threads;
-                return true;
+
```

Comment:
The final value to be used should be restricted to the max possible count (short.MaxValue)